.github/workflows/docker-build.yml

```diff
@@ -39,7 +39,8 @@ jobs:
           done
   build:
     needs: define-matrix
-    runs-on: ubuntu-latest
+    runs-on: >-
+      ${{ (contains(matrix.component, 'arm64')) && 'ubuntu-24.04-arm' || 'ubuntu-latest' }}
```

Comment:
How about the following simple solution using just the `matrix` feature?

```yaml
strategy:
  fail-fast: false
  matrix:
    component: ${{ fromJSON(needs.define-matrix.outputs.components) }}
    include:
      - runs-on: ubuntu-latest
      - runs-on: ubuntu-24.04-arm
        component: arm64
runs-on: ${{ matrix.runs-on }}
```

ref: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#expanding-or-adding-matrix-configurations

Reply:
Sorry, I missed the backport of this fix, which has already been merged.

* #417

We should consider this point separately later.